distributed/client.py

```diff
@@ -578,14 +597,16 @@ def start(self, asynchronous=None, **kwargs):
                 self._should_close_loop = True
             while not self.loop._running:
                 sleep(0.001)
+
         pc = PeriodicCallback(lambda: None, 1000, io_loop=self.loop)
         self.loop.add_callback(pc.start)
         _set_global_client(self)
-        if asynchronous:
+        self.status = 'connecting'
+
+        if self.asynchronous:
```

Comment:
There is some value to reducing the number of active clients.  I agree though that, especially because of the failure of synchronous-intended clients to work in asynchronous environments, we do need to resolve this.  

I think I'm still hoping for a way to help identify whether we should or should not act in an asynchronous manner that is separate from defining a client as asynchronous or not.

My inclination here is to merge things roughly as they are now.  I think that the user-facing bits of this are decent.  We have a known failure in one case but it's unlikely to be encountered by most users.  This PR is large and I'd like to get it in soon-ish.  I don't have strong confidence that I can find a good solution in the next day or two with other things going on.